src/pcre2_jit_compile.c

```diff
@@ -13887,14 +13886,18 @@ if (has_alternatives)
   if (cond != NULL)
     {
     SLJIT_ASSERT(opcode == OP_COND || opcode == OP_SCOND);
-    assert = CURRENT_AS(bracket_backtrack)->u.assert;
-    if ((ccbegin[1 + LINK_SIZE] == OP_ASSERT_NOT || ccbegin[1 + LINK_SIZE] == OP_ASSERTBACK_NOT) && assert->framesize >= 0)
+    if (ccbegin[1 + LINK_SIZE] == OP_ASSERT_NOT || ccbegin[1 + LINK_SIZE] == OP_ASSERTBACK_NOT)
       {
-      OP1(SLJIT_MOV, STACK_TOP, 0, SLJIT_MEM1(SLJIT_SP), assert->private_data_ptr);
-      add_jump(compiler, &common->revertframes, JUMP(SLJIT_FAST_CALL));
-      OP1(SLJIT_MOV, TMP1, 0, SLJIT_MEM1(STACK_TOP), STACK(-2));
-      OP2(SLJIT_ADD, STACK_TOP, 0, STACK_TOP, 0, SLJIT_IMM, (assert->framesize - 1) * sizeof(sljit_sw));
-      OP1(SLJIT_MOV, SLJIT_MEM1(SLJIT_SP), assert->private_data_ptr, TMP1, 0);
+      assert = CURRENT_AS(bracket_backtrack)->u.assert;
+      if (assert->framesize > 1)
+        {
+        OP1(SLJIT_MOV, STACK_TOP, 0, SLJIT_MEM1(SLJIT_SP), assert->private_data_ptr);
+        add_jump(compiler, &common->revertframes, JUMP(SLJIT_FAST_CALL));
+        OP1(SLJIT_MOV, TMP1, 0, SLJIT_MEM1(STACK_TOP), STACK(-2));
+        OP2(SLJIT_ADD, STACK_TOP, 0, STACK_TOP, 0, SLJIT_IMM, (assert->framesize - 1) * sizeof(sljit_sw));
+        OP1(SLJIT_MOV, SLJIT_MEM1(SLJIT_SP), assert->private_data_ptr, TMP1, 0);
+        }
+      else if (assert->framesize == 0) SLJIT_UNREACHABLE();
```

Comment:
put the `SLJIT_ASSERT(assert->framesize > 0);` after the `assert = CURRENT_AS(bracket_backtrack)->u.assert;`.

Reply:
sorry; I was sleep deprived and really meant to use `PCRE2_UNREACHABLE()` here and only as a last resort to catch any more of those heisenbugs before removing the asserts completely, since they IMHO no longer make sense because the code explicitly filters out any framesize below 1.

will put the asserts back in their "normal" place as you requested though, but still think we are better without them.

Reply:
I recommend to sleep, this is definitely not urgent.

Reply:
I know, it was just the frustration of seeing such a simple change cause crashes, and the shame of not caching those (twice) in local testing; plus my unfamiliarity with the code that required a lot of reading.

luckily I found another 10 year old "bug", so definitely worth it.